File: python/project-euler/PE019.py
"""
How many Sundays fell on the first of the month during the 
twentieth century (1 Jan 1901 to 31 Dec 2000)?
"""

import datetime


class PE019:
  def count_sundays(self):
    sundays = 0
    for year in range(1901, 2000+1):
      for month in range(1, 12+1):
        if datetime.datetime(year, month, 1).isoweekday() == 7:
          sundays += 1
    return sundays


# test
test = PE019()
print(test.count_sundays())  # 171

File: java/project-euler/PE019.java

/*
 * How many Sundays fell on the first of the month during the twentieth century
 * (1 Jan 1901 to 31 Dec 2000)?
 */

import java.util.Calendar;

public class PE019 {
  public int countSundays() {
    Calendar cal = Calendar.getInstance();
    int sundays = 0;
    for (int year = 1901; year <= 2000; year++) {
      for (int month = 0; month < 12; month++) {
        cal.set(year, month, 1);
        if (cal.get(Calendar.DAY_OF_WEEK) == Calendar.SUNDAY) {
          sundays++;
        }
      }
    }
    return sundays;
  }

  public static void main(String[] args) {
    PE019 test = new PE019();
    System.out.println(test.countSundays()); // 173
  }
}